Sources/Controllers/MessagesViewController.swift

```diff
@@ -259,17 +264,33 @@ extension MessagesViewController {
 
         if (keyboardEndFrame.origin.y + keyboardEndFrame.size.height) > UIScreen.main.bounds.height {
             // Hardware keyboard is found
-            let bottomInset = view.frame.size.height - keyboardEndFrame.origin.y
+            let bottomInset = view.frame.size.height - keyboardEndFrame.origin.y - iPhoneXBottomInset()
             messagesCollectionView.contentInset.bottom = bottomInset
             messagesCollectionView.scrollIndicatorInsets.bottom = bottomInset
 
         } else {
             //Software keyboard is found
-            let bottomInset = keyboardEndFrame.height > (inputAccessoryView?.frame.height ?? 0) ? keyboardEndFrame.height : (inputAccessoryView?.frame.height ?? 0)
+            let bottomInset = keyboardEndFrame.height > keyboardOffsetFrame().height ? (keyboardEndFrame.height - iPhoneXBottomInset()) : keyboardOffsetFrame().height
             messagesCollectionView.contentInset.bottom = bottomInset
             messagesCollectionView.scrollIndicatorInsets.bottom = bottomInset
         }
         
     }
     
+    fileprivate func keyboardOffsetFrame() -> CGRect {
```

Comment:
We can also make this a computed property

Reply:
ya